lib/ngx/log.md

```diff
@@ -0,0 +1,231 @@
+Name
+====
+
+`ngx.log` - manage the nginx error log for OpenResty/ngx_lua.
+
+Table of Contents
+=================
+
+* [Name](#name)
+* [Status](#status)
+* [Synopsis](#synopsis)
+    * [Intercept nginx error logs with specified log level](#intercept-nginx-error-logs-with-specified-log-level)
+* [Methods](#methods)
+    * [set_errlog_filter](#set_errlog_filter)
+    * [get_errlog](#get_errlog)
+* [Community](#community)
+    * [English Mailing List](#english-mailing-list)
+    * [Chinese Mailing List](#chinese-mailing-list)
+* [Bugs and Patches](#bugs-and-patches)
+* [Author](#author)
+* [Copyright and License](#copyright-and-license)
+* [See Also](#see-also)
+
+Status
+======
+
+This Lua module is currently considered experimental.
+The API is still in flux and may change in the future without notice.
+
+Synopsis
+========
+
+Intercept nginx error logs with specified log level
+-----------------------------------------
+
+```nginx
+error logs/error.log info;
+
+http {
+    # enable intercept error log
+    lua_intercept_error_log 32m;
+
+    init_by_lua_block {
+        local ngx_log = require "ngx.log"
+        local status, err = ngx_log.set_errlog_filter(ngx.WARN)
+        if not status then
+            ngx.log(ngx.ERR, err)
+            return
+        end
+        ngx.log(ngx.WARN, "set error filter level: WARN")
+    }
+
+    server {
+        # ...
+        location = /t {
+            content_by_lua_block {
+                local ngx_log = require "ngx.log"
+                ngx.log(ngx.INFO, "test1")
+                ngx.log(ngx.WARN, "test2")
+                ngx.log(ngx.ERR, "test3")
+
+                local logs, err = ngx_log.get_errlog()
+                if not logs then
+                    ngx.say("FAILED ", err)
+                    return
+                end
+
+                for _, log in ipairs(logs) do
+                    ngx.say("level: ", log[1], " data: ", log[2])
+                end
+            }
+        }
+    }
+}
+
+```
+
+The example location above produces a response like this:
+
+```
+level: 5 data: 2017/04/19 22:20:03 [warn] 63176#0:
+    [lua] init_by_lua:7: set error filter level: WARN
+level: 5 data: 2017/04/19 22:20:05 [warn] 63176#0: *1
+    [lua] content_by_lua(nginx.conf:59):4: test2
+level: 4 data: 2017/04/19 22:20:05 [error] 63176#0: *1
+    [lua] content_by_lua(nginx.conf:59):5: test3
+```
+
+[Back to TOC](#table-of-contents)
+
+Methods
+=======
+
+set_errlog_filter
+---
+**syntax:** *status, err = log_module.set_errlog_filter(log_level)*
+
+**context:** *init_by_lua&#42;*
+
+Specify the filter log level, only to intercept the error log we need.
+If we don't call this API, all of the error logs will be intercepted by default.
```

Comment:
Better add a note about debugging logs since when OpenResty or NGINX is not built with `--with-debug`, all the debug level logs are suppressed regardless.